src/Microsoft.SqlTools.ServiceLayer/SqlAssessment/SqlAssessmentService.cs

```diff
@@ -0,0 +1,463 @@
+        /// registers request handlers and shutdown event handler.
+        /// </summary>
+        /// <param name="serviceHost">The service host instance to register with</param>
+        public void InitializeService(ServiceHost serviceHost)
+        {
+            // Register handlers for requests
+            serviceHost.SetRequestHandler(InvokeRequest.Type, HandleInvokeRequest);
+            serviceHost.SetRequestHandler(GetAssessmentItemsRequest.Type, HandleGetAssessmentItemsRequest);
+
+            // Register handler for shutdown event
+            serviceHost.RegisterShutdownTask((shutdownParams, requestContext) =>
+            {
+                Dispose();
+                return Task.FromResult(0);
+            });
+        }
+
+        #region Request Handlers
+
+        /// <summary>
+        /// This function obtains a live connection, then calls
+        /// an assessment operation specified by <paramref name="assessmentFunc"/>
+        /// </summary>
+        /// <typeparam name="TResult">
+        /// SQL Assessment result item type.
+        /// </typeparam>
+        /// <param name="requestParams">
+        /// Request parameters passed from the host.
+        /// </param>
+        /// <param name="connectParams">
+        /// Connection parameters used to identify and access the target.
+        /// </param>
+        /// <param name="taskUri">
+        /// An URI identifying the request task to enable concurrent execution.
+        /// </param>
+        /// <param name="assessmentFunc">
+        /// A function performing assessment operation for given target.
+        /// </param>
+        /// <returns>
+        /// Returns <see cref="AssessmentResult{TResult}"/> for given target.
+        /// </returns>
+        internal async Task<AssessmentResult<TResult>> CallAssessmentEngine<TResult>(
+            AssessmentParams requestParams,
+            ConnectParams connectParams,
+            string taskUri,
+            Func<SqlObjectLocator, Task<List<TResult>>> assessmentFunc)
+        {
+            var result = new AssessmentResult<TResult>
+                             {
+                                 RulesetVersion = Engine.Configuration.DefaultRuleset.Version.ToString(),
+                                 ApiVersion = "1.0"
+                             };
+
+            await this.ConnectionService.Connect(connectParams);
+            var connection = await ConnectionService.Instance.GetOrOpenConnection(taskUri, ConnectionType.Query);
+
+            try
+            {
+                if (connection.State == ConnectionState.Closed)
+                {
+                    connection.Open();
+                }
+
+                // Obtain target server details
+                var cmd = ((DbConnection)connection).CreateCommand();
+                var server = new SqlObjectLocator();
+
+                const string PlatformQueryText = @"SELECT [host_platform] FROM sys.dm_os_host_info";

```

Comment:
Assuming these queries are actually necessary for the assessment request you should use a common helper like ReliableConnectionHelper.GetServerVersion for this.

Reply:
I had to extend the helper to return ServerName.